In [2]:
#Step-1
from sklearn.datasets import make_classification
data_set = make_classification(n_samples=10000, n_features=20, n_informative=10, n_redundant=2, n_repeated=0, n_classes=2)

In [3]:
data_set

(array([[ 5.60106135,  4.89727919,  0.31129903, ...,  0.30747077,
          0.5703885 , -1.58379417],
        [-1.27263917, -2.50471916,  1.01919819, ...,  1.17697855,
          0.01394869,  3.10053093],
        [ 0.48759804, -2.19540298,  0.97933453, ...,  1.16918413,
          0.71846281,  0.68028492],
        ...,
        [-0.56449393,  0.56970075,  0.26144779, ..., -1.22759919,
          0.6497699 ,  2.31388283],
        [-0.4354978 ,  4.73396776, -0.15308976, ...,  0.59582748,
         -1.23768262, -0.89767465],
        [ 0.39978934,  1.11432727,  1.5257167 , ..., -4.00700833,
         -1.82980621,  0.95694615]]), array([0, 1, 1, ..., 1, 0, 1]))

In [4]:
#Step-2
from sklearn.neural_network import MLPClassifier

In [5]:
import numpy as np
from sklearn.model_selection import KFold

In [6]:
X = data_set[0]
y = data_set[1]
print(X)
print(y)

[[ 5.60106135  4.89727919  0.31129903 ...  0.30747077  0.5703885
  -1.58379417]
 [-1.27263917 -2.50471916  1.01919819 ...  1.17697855  0.01394869
   3.10053093]
 [ 0.48759804 -2.19540298  0.97933453 ...  1.16918413  0.71846281
   0.68028492]
 ...
 [-0.56449393  0.56970075  0.26144779 ... -1.22759919  0.6497699
   2.31388283]
 [-0.4354978   4.73396776 -0.15308976 ...  0.59582748 -1.23768262
  -0.89767465]
 [ 0.39978934  1.11432727  1.5257167  ... -4.00700833 -1.82980621
   0.95694615]]
[0 1 1 ... 1 0 1]


In [7]:
from sklearn.model_selection import cross_val_score
l = [(5,),(5,5,),(5,10,),(10,),(10,5,),(10,10,)]
for fun in ['logistic']:
    for layers in l:
        mlp=MLPClassifier(hidden_layer_sizes=layers,activation=fun,max_iter=3000)
        mlp.fit(X,y)
        scores=cross_val_score(mlp, X, y, cv=5)
        print("Activation Function :",fun,"Layer Shape :",str(layers),"Cross Validation Score :",round(np.average(scores),3))

Activation Function : logistic Layer Shape : (5,) Cross Validation Score : 0.985
Activation Function : logistic Layer Shape : (5, 5) Cross Validation Score : 0.987
Activation Function : logistic Layer Shape : (5, 10) Cross Validation Score : 0.988
Activation Function : logistic Layer Shape : (10,) Cross Validation Score : 0.985
Activation Function : logistic Layer Shape : (10, 5) Cross Validation Score : 0.989
Activation Function : logistic Layer Shape : (10, 10) Cross Validation Score : 0.99


Here we could see better score for NN of shape (10,). Hence we are holding this network shape

In [10]:
#step3
data_set1 = make_classification(n_samples=10000, n_features=20, n_informative=5, n_redundant=2, n_repeated=0, n_classes=2)
X = data_set1[0]
y = data_set1[1]
for fun in ['logistic']:
    for layers in [(10,)]:
        mlp=MLPClassifier(hidden_layer_sizes=layers,activation=fun,max_iter=3000)
        mlp.fit(X,y)
        scores=cross_val_score(mlp, X, y, cv=5)
        print("Activation Function :",fun,"Layer Shape :",str(layers),"Cross Validation Score :",round(np.average(scores),3))

Activation Function : logistic Layer Shape : (10,) Cross Validation Score : 0.945


Cross Val Score has been slightly reduced to 94.5% when n_informative is reduced from 10 to 5

In [9]:
data_set2 = make_classification(n_samples=10000, n_features=20, n_informative=3, n_redundant=2, n_repeated=0, n_classes=2)
X = data_set2[0]
y = data_set2[1]
for fun in ['logistic']:
    for layers in [(10,)]:
        mlp=MLPClassifier(hidden_layer_sizes=layers,activation=fun,max_iter=3000)
        mlp.fit(X,y)
        scores=cross_val_score(mlp, X, y, cv=5)
        print("Activation Function :",fun,"Layer Shape :",str(layers),"Cross Validation Score :",round(np.average(scores),3))

Activation Function : logistic Layer Shape : (10,) Cross Validation Score : 0.943


Cross Val Score has been slightly reduced to 94.3% when n_informative is reduced from 10 to 3